libs/common/RespReadUtils.cs

```diff
@@ -1040,5 +923,171 @@ public static bool ReadSerializedData(out byte[] key, out byte[] value, out long
 
             return true;
         }
+
+        #region ClientSpecific
+        /// <summary>
+        /// Try to read a RESP formatted bulk string
+        /// NOTE: This is used with client implementation to parse responses that may include a null value (i.e. $-1\r\n)
+        /// </summary>
+        /// <param name="result"></param>
+        /// <param name="ptr"></param>
+        /// <param name="end"></param>
+        /// <returns></returns>
+        public static bool ReadStringResponseWithLengthHeader(out string result, ref byte* ptr, byte* end)
```

Comment:
Looks pretty similar to the next overload except for the setting of result, maybe extract the logic to a helper function?

Reply:
I updated this too in the latest commit. Please take a look.